pkg/volume/secret/secret.go

```diff
@@ -193,10 +193,14 @@ func (b *secretVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 		return fmt.Errorf("Cannot setup secret volume %v because kube client is not configured", b.volName)
 	}
 
+	optional := b.source.Optional != nil && *b.source.Optional
 	secret, err := kubeClient.Core().Secrets(b.pod.Namespace).Get(b.source.SecretName, metav1.GetOptions{})
 	if err != nil {
-		glog.Errorf("Couldn't get secret %v/%v", b.pod.Namespace, b.source.SecretName)
-		return err
+		if !optional {
+			glog.Errorf("Couldn't get secret %v/%v", b.pod.Namespace, b.source.SecretName)
+			return err
+		}
+		secret = &v1.Secret{}
```

Comment:
probably want to at least populate the name/namespace of the stub object so any logging that uses it later will log helpfully